### PR TITLE
fix: set limitnofile

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:
https://d2iq.atlassian.net/browse/D2IQ-96107
This change in contained seems to cause problems for workloads that use disk https://github.com/containerd/containerd/pull/4475
**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by dkoshkin — APPROVED** [review]

**Review by mkarroqe — APPROVED** [review]

**Review by supershal — APPROVED** [review]

**Comment by dlipovetsky** [comment]
When we upgraded containerd 1.4 to 1.6, the systemd unit changed the file descriptor limit from `1024768` to `infinity`. This PR effectively returns the limit to what it was in DKP 2.4 and earlier.

**Review by dlipovetsky — APPROVED** [review]

**Comment by faiq** [comment]
Looking upstream this seems to be caused by [this PR](https://github.com/containerd/containerd/pull/4475/files) which sets the containerd limits to unlimited, users who were previously on 1.4.13 version of containerd provided by dkp 2.4 and previous had this limit set. Removing this limit caused our rook deployment to break and we also saw problem with HA proxy as well which had this limit as the root cause. We decided to bring this limit back for a couple of reasons

1. We don’t know if this can break existing users applications 
2. It breaks our application

**Comment by dkoshkin** [comment]
@faiq I think we are good to merge this, the 2 other fixes look good:

* [RHSM vars](https://github.com/mesosphere/konvoy-image-builder/actions/runs/4325203880/jobs/7551054047#step:9:2455)
* [vSphere dry-run](https://github.com/mesosphere/konvoy-image-builder/actions/runs/4325203867/jobs/7551050473#step:9:3692)

File: ansible/roles/containerd/tasks/main.yaml [hunk 1]
```diff
@@ -4,6 +4,12 @@
     path: "/etc/systemd/system/containerd.service.d/"
     state: directory
 
+- name: Create containerd max files
+  template:
+    dest: /etc/systemd/system/containerd.service.d/max-files.conf
+    src: etc/systemd/system/containerd.service.d/max-files.conf
+    mode: 0644
```
Comment (fatz) [inline ansible/roles/containerd/tasks/main.yaml:11]: @faiq I started to rebase my molecule test PR #553 

it turns out that this part fails with Ubuntu. This is when `http_proxy` or `https_proxy`is not set and the first task is not run due to that. 

My suggestion:

Move this part after install includes and remove the conditions from directory creation

Comment (fatz) [reply ansible/roles/containerd/tasks/main.yaml:11]: @faiq cherry-picking 137be77639a90cb75cdc3d55831404e02b85823a should help

**Pull request merged**